text.cpp

```diff
@@ -91,7 +91,16 @@ class Text {
       }
 
       auto isMatchingSymbol = [&](const Symbol& symbol) -> bool {
-          return symbol.id == static_cast<unsigned char>(c);
+        /* c is probably char_t here, so ...
+          casting first to unsigned char is necessary, because if we
+          cast to int32_t directly, the machine bit extends, meaning
+          that the match would fail because of the type's sign.
+          we could also do uint32_t, since I think zengl supports utf-32
+          but I don't think that's necessary because we are not support 
+          unicode anyway (i.e. we never go beyond 0xFF)
```

Comment:
depending on what zengl uses we should fix the zengl struct and also this cast. I stole my zengl struct from the font tool source and changed some datatypes to stdint iirc.